code/modules/shuttle/navigation_computer.dm

```diff
@@ -0,0 +1,379 @@
+		current_user.client.images -= the_eye.placed_images
+
+	QDEL_LIST(the_eye.placed_images)
+
+	for(var/V in the_eye.placement_images)
+		var/image/I = V
+		var/image/newI = image('icons/effects/alphacolors.dmi', the_eye.loc, "blue")
+		newI.loc = I.loc //It is highly unlikely that any landing spot including a null tile will get this far, but better safe than sorry.
+		newI.layer = ABOVE_OPEN_TURF_LAYER
+		newI.plane = 0
+		newI.mouse_opacity = 0
+		the_eye.placed_images += newI
+
+	if(current_user.client)
+		current_user.client.images += the_eye.placed_images
+		to_chat(current_user, "<span class='notice'>Transit location designated.</span>")
+	return TRUE
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/proc/canDesignateTarget()
+	if(!designating_target_loc || !current_user || (eyeobj.loc != designating_target_loc) || (machine_stat & (NOPOWER|BROKEN)) )
+		return FALSE
+	return TRUE
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/proc/rotateLandingSpot()
+	var/mob/camera/aiEye/remote/shuttle_docker/the_eye = eyeobj
+	var/list/image_cache = the_eye.placement_images
+	the_eye.setDir(turn(the_eye.dir, -90))
+	for(var/i in 1 to image_cache.len)
+		var/image/pic = image_cache[i]
+		var/list/coords = image_cache[pic]
+		var/Tmp = coords[1]
+		coords[1] = coords[2]
+		coords[2] = -Tmp
+		pic.loc = locate(the_eye.x + coords[1], the_eye.y + coords[2], the_eye.z)
+	var/Tmp = x_offset
+	x_offset = y_offset
+	y_offset = -Tmp
+	checkLandingSpot()
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/proc/checkLandingSpot()
+	var/mob/camera/aiEye/remote/shuttle_docker/the_eye = eyeobj
+	var/turf/eyeturf = get_turf(the_eye)
+	if(!eyeturf)
+		return SHUTTLE_DOCKER_BLOCKED
+	if(!eyeturf.z || SSmapping.level_has_any_trait(eyeturf.z, locked_traits))
+		return SHUTTLE_DOCKER_BLOCKED
+
+	. = SHUTTLE_DOCKER_LANDING_CLEAR
+	var/list/bounds = shuttle_port.return_coords(the_eye.x - x_offset, the_eye.y - y_offset, the_eye.dir)
+	var/list/overlappers = SSshuttle.get_dock_overlap(bounds[1], bounds[2], bounds[3], bounds[4], the_eye.z)
+	var/list/image_cache = the_eye.placement_images
+	for(var/i in 1 to image_cache.len)
+		var/image/I = image_cache[i]
+		var/list/coords = image_cache[I]
+		var/turf/T = locate(eyeturf.x + coords[1], eyeturf.y + coords[2], eyeturf.z)
+		I.loc = T
+		switch(checkLandingTurf(T, overlappers))
+			if(SHUTTLE_DOCKER_LANDING_CLEAR)
+				I.icon_state = "green"
+			if(SHUTTLE_DOCKER_BLOCKED_BY_HIDDEN_PORT)
+				I.icon_state = "green"
+				if(. == SHUTTLE_DOCKER_LANDING_CLEAR)
+					. = SHUTTLE_DOCKER_BLOCKED_BY_HIDDEN_PORT
+			else
+				I.icon_state = "red"
+				. = SHUTTLE_DOCKER_BLOCKED
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/proc/checkLandingTurf(turf/T, list/overlappers)
+	// Too close to the map edge is never allowed
+	if(!T || T.x <= 10 || T.y <= 10 || T.x >= world.maxx - 10 || T.y >= world.maxy - 10)
+		return SHUTTLE_DOCKER_BLOCKED
+	// If it's one of our shuttle areas assume it's ok to be there
```

Comment:
Around here, check the turfs area, and include a `SHUTTLE_DOCKER_BLOCKED` for when the `ceiling` var of the area is `>= CEILING_DEEP_UNDERGROUND`

Reply:
It has been done... and it works.